### DISCUSSION TITLE
AMI diarization

### DISCUSSION THREAD

**Discussion by someonewating** [speechbrain/speechbrain#1943] [Q&A]
Hi there. I'm following the recipe for speaker diarization on the AMI dataset(https://github.com/speechbrain/speechbrain/tree/develop/recipes/AMI/Diarization). I got the totally same error as #1056, which is: 
```
__main__ - Tuning for p-value for SC (Multiple iterations over AMI Dev set)
__main__ - Diarizing dev set
__main__ - No recording IDs found! Please check if meta_data json file is properly generated.
```

However, I'm 100% sure I'm using correct path. As mentioned in #1056, I'm using the full path to the folders. So I don't know why I can't run the experiment.py successfully.

My path in `xvectors.yaml` is 
```
data_folder: /home/john/dp/resources/ami/amicorpus/ # e.g., /path/to/amicorpus/

manual_annot_folder: /home/john/dp/resources/ami/ami_public_manual_1.6.2/ # e.g., /path/to/ami_public_manual_1.6.2/

output_folder: /home/john/dp/speechbrain/git/recipes/AMI/Diarization/results/ami/xvect/
save_folder: !ref <output_folder>/save/
```

My `ami_dev.Mix-Headset.subsegs.json` and `ami_eval.Mix-Headset.subsegs.json` are empty too, while my `ami_train.Mix-Headset.subsegs.json` looks like: 
```
{
  "ES2008a_10.462_13.406": {
    "wav": {
      "file": "/home/john/dp/resources/ami/amicorpus//ES2008a/audio/ES2008a.Mix-Headset.wav",
      "duration": 2.944,
      "start": 167392,
      "stop": 214496
    }
  },
  "ES2008a_27.863_28.533": {
    "wav": {
      "file": "/home/john/dp/resources/ami/amicorpus//ES2008a/audio/ES2008a.Mix-Headset.wav",
      "duration": 0.67,
      "start": 445808,
      "stop": 456528
    }
  },
  ...
```
In my home directory, I typed command `python3 /home/john/dp/speechbrain/git/recipes/AMI/Diarization/experiment.py /home/john/dp/speechbrain/git/recipes/AMI/Diarization/hparams/xvectors.yaml`, and I got the result: 
```
torchvision is not available - cannot save figures
speechbrain.core - Beginning experiment!
speechbrain.core - Experiment folder: /home/john/dp/speechbrain/git/recipes/AMI/Diarization/results/ami/xvect/
speechbrain.pretrained.fetching - Fetch embedding_model.ckpt: Using existing file/symlink in /home/john/dp/speechbrain/git/recipes/AMI/Diarization/results/ami/xvect/save/embedding_model.ckpt.
speechbrain.pretrained.fetching - Fetch mean_var_norm_emb.ckpt: Using existing file/symlink in /home/john/dp/speechbrain/git/recipes/AMI/Diarization/results/ami/xvect/save/mean_var_norm_emb.ckpt.
speechbrain.utils.parameter_transfer - Loading pretrained files for: embedding_model, mean_var_norm_emb
__main__ - Tuning for p-value for SC (Multiple iterations over AMI Dev set)
__main__ - Diarizing dev set
__main__ - No recording IDs found! Please check if meta_data json file is properly generated.
```

I'll be appreciated for any help!

**Comment by someonewating** [comment]
@sch0ngut Would you mind explaining your solution more specifically? 
@LONG520520 Would you mind providing a new template? The previous can't work right now.
Thank you.

**Reply by sch0ngut** [reply]
Hey @someonewating! As explained in the thread you linked, the way I made it work in the end was to play with the paths, without actually really understanding why some things worked and some things did not work. I'm sorry that I can't give any more detail. Best of luck!
